java/src/main/java/svix/Webhook.java

```diff
@@ -0,0 +1,62 @@
+package svix;
+
+import java.net.http.HttpHeaders;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Optional;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+public class Webhook {
+	private static final String HMAC_SHA512 = "HmacSHA512";
+
+	private final byte[] key;
+
+	public Webhook(String secret) {
+		this.key = Base64.getDecoder().decode(secret);
+	}
+
+	public String verify(String payload, HttpHeaders headers) throws WebhookVerificationError {
+		Optional<String> msgId = headers.firstValue("svix-id");
+		Optional<String> msgSignature = headers.firstValue("svix-signature");
+		Optional<String> msgTimestamp = headers.firstValue("svix-timestamp");
+
+		if (msgId.isEmpty() || msgSignature.isEmpty() || msgTimestamp.isEmpty()) {
+			throw new WebhookVerificationError("Missing required headers");
+		}
+
+		String toSign = String.format("%s.%s.%s", msgId, msgSignature, msgTimestamp);
+		String expectedSignature = sign(toSign);
+		String[] passedSignatures = expectedSignature.split(" ");
+		for (String versionedSignature : passedSignatures) {
+			String[] sigParts = versionedSignature.split(",");
+			if (sigParts.length < 2) {
+				continue;
+			}
+			String version = sigParts[0];
+			String signature = sigParts[1];
+			if (!version.equals("v1")) {
+				continue;
+			}
+			if (signature.equals(expectedSignature)) {
+				return new JSON().serialize(payload);
```

Comment:
That's a good question. I'm not using JSON nowadays so I'm not sure what most people are used to. What I did is actually wrong.

I can think of 3 options here, not sure what is the best approach that will be consistent with the other languages.
1. Just return the string as is (a bit weird, as we accept it as an argument and don't use it)
2. Change the method to return boolean
3. If the payload can be represented with an object then return that object

what do you think?

Reply:
I personally think its best to just return a bool (or really nothing and just throw an error) and punt on handling the JSON marshaling and punt that to the user, This is how I did it in the go library and will probably do going forward.  Parsing made sense as a convince in the javascript library but since the payload is arbitrary it really doesn't make sense for us to do it here.

@tasn do you agree?

Reply:
Yeah, I'm 100% with Frank on this one. Just throw an error on failure and return nothing on success.

P.s, @talhacohen, you are definitely using the string though, for the verification of the signature (or at least should, haha).

Reply:
True, I'm using it, changed to boolean then.
I can also remove all the exceptions here and just return false when it's not valid. for now, keeping the exception for consistency with go.

This code isn't working yet (at least the tests are failing, still looking into it)

Reply:
Done